File: Python/121.py
class Solution(object):
    def maxProfit(self, prices):
        Max = 0
        total = 0
        Min = prices[0]
        for i in range(1,len(prices)):
            total = prices[i] - Min
            if total > Max:
                Max = total
            if prices[i] < Min:
                Min = prices[i]
        return Max

File: Java/121.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0, min = prices[0],total = 0;
        for(int i = 0; i < prices.length; i++){
            total = prices[i] - min;
            max = Math.max(max,total);
            if(prices[i] < min) min = prices[i];
        }
        return max;
    }
}